CRM/CiviAwards/Service/AwardProfile.php

```diff
@@ -53,7 +63,25 @@ public function createProfileFields(array $customFields, $profileId) {
     $this->addContactIdField($profileId);
   }
 
-    $this->addContactIdField($profileId);
+  /**
+   * Validates review_field options to ensure legal options are passed.
+   *
+   * @param array
+   *  Custom field
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  private function validateReviewFields($customField) {
```

Comment:
Also, we need to move the validation logic to the AwardDetails BAO. See: https://github.com/compucorp/uk.co.compucorp.civiawards/blob/RSE-44-awards-management/CRM/CiviAwards/BAO/AwardDetail.php#L59
The reason is because
(1)  it's better to have all the validations for an Award details in one place
(2) If we have it here, it is possible to create an Award detail successfully and the Award Profile creation will fail because of a validation error since this is an after save of the Award detail creation.